### PR TITLE
Android: update gradle wrapper & cmake

### PR DESCRIPTION
**Pull request by PatriceJiang**
Resolve: https://github.com/cocos/3d-tasks/issues/16261

https://github.com/cocos/cocos-engine/issues/14964

This PR updates the project's build tools and configurations:

The purpose of these upgrades is to take advantage of the latest features, improvements, and bug fixes provided by the updated build tools.

### Changelog

- Upgraded Gradle Wrapper from version 6.5.1 to 7.5.0
- Upgraded Android Gradle Plugin from version 4.1.0 to 7.4.0
- Specified CMake version 3.22.1
- Updated necessary configurations to align with the new versions

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
@cocos-robot run test case

**Comment by minggo** [comment]
Will it break compatibility?

File: templates/android/build/build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         // jcenter() // keeped as anchor, will be removed soon
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.0'
+        classpath 'com.android.tools.build:gradle:7.4.0'
```
Comment (PatriceJiang) [inline templates/android/build/build.gradle]:
```suggestion
        classpath 'com.android.tools.build:gradle:7.4.2'
```

**Review by dumganhar — APPROVED** [review]

**Comment by PatriceJiang** [comment]
> Will it break compatibility?

It will not affect compatibility, and older projects can still be compiled using the older version of Android Studio.

**Comment by minggo** [comment]
> > Will it break compatibility?
> 
> It will not affect compatibility, and older projects can still be compiled using the older version of Android Studio.

What will happen if use older Android Studio with new template?

**Comment by PatriceJiang** [comment]
> > > Will it break compatibility?
> > 
> > 
> > It will not affect compatibility, and older projects can still be compiled using the older version of Android Studio.
> 
> What will happen if use older Android Studio with new template?

It can be used normally as Gradle does not depend on the version of Android Studio, but there may be prompts to upgrade.

**Review by minggo — APPROVED** [review]

**Pull request merged**